mender-deb-package

```diff
@@ -55,9 +55,9 @@ checkout_repo_clean_local_path() {
 
 checkout_repo() {
   local -r repo_path=$(checkout_repo_clean_local_path "${REPO_URL}")
-  git clone ${REPO_URL} ${repo_path}
+  git clone --depth=1 --branch ${VERSION} ${REPO_URL} ${repo_path}
```

Comment:
Careful, this won't work with tags :boom: 

Either leave it as it was (full clone, then checkout) or use the mender-qa way (inspiration [here](https://github.com/mendersoftware/mender-qa/blob/f9dbfd797299aff75da060128012edd667d2c74e/gitlab-pipeline/stage/init.yml#L54))

Reply:
I'm guessing the submodules can be kept at `depth=1` though

Reply:
Hmm, the docs says it should work just fine with tags as well :thinking: 

```
--branch <name>

    Instead of pointing the newly created HEAD to the branch pointed to by the cloned repository’s HEAD, point to <name> branch instead. In a non-bare repository, this is the branch that will be checked out. --branch can also take tags and detaches the HEAD at that commit in the resulting repository.
```

Am I missing something obvious?

Reply:
Resolved, this works